.github/workflows/ci.yaml

```diff
@@ -209,162 +139,202 @@ jobs:
+            DEVELOPER: 1
+            EXPERIMENTAL_FEATURES: 1
+            TEST_DB_PROVIDER: sqlite3
+            COMPILER: clang
+          # And of course we want to test postgres too
+          - NAME: postgres
+            CFG: gcc-dev1-exp1
+            DEVELOPER: 1
+            EXPERIMENTAL_FEATURES: 1
+            COMPILER: gcc
+            TEST_DB_PROVIDER: postgres
+
     steps:
       - name: Checkout
-        uses: actions/checkout@v2.0.0
+        uses: actions/checkout@v3
 
       - name: Set up Python 3.7
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: 3.7
 
       - name: Install dependencies
         run: |
-          bash -x .github/scripts/setup.sh
+          pip3 install --user pip wheel poetry
+          poetry install
 
-      - name: Build
+      - name: Install bitcoind
+        run: .github/scripts/install-bitcoind.sh
+
+      - name: Download build
+        uses: actions/download-artifact@v3
+        with:
+          name: cln-${{ matrix.CFG }}.tar.bz2
+
+      - name: Test
         env:
           VALGRIND: ${{ matrix.VALGRIND }}
           DEVELOPER: ${{ matrix.DEVELOPER }}
           EXPERIMENTAL_FEATURES: ${{ matrix.EXPERIMENTAL_FEATURES }}
-          EXPERIMENTAL_DUAL_FUND: ${{ matrix.EXPERIMENTAL_DUAL_FUND }}
           COMPILER: ${{ matrix.COMPILER }}
-          ARCH: ${{ matrix.ARCH }}
-          COMPAT: ${{ matrix.COMPAT }}
-          PYTEST_PAR: ${{ matrix.PYTEST_PAR }}
-          PYTEST_OPTS: ${{ matrix.PYTEST_OPTS }}
-          NETWORK: ${{ matrix.NETWORK }}
-          TEST_CMD: ${{ matrix.TEST_CMD }}
-          TEST_GROUP_COUNT: ${{ matrix.TEST_GROUP_COUNT }}
-          TEST_GROUP: ${{ matrix.TEST_GROUP }}
+          COMPAT: 1
+          CFG: ${{ matrix.CFG }}
+          SLOW_MACHINE: 1
+          PYTEST_PAR: 10
+          TEST_DEBUG: 1
+          TEST_DB_PROVIDER: ${{ matrix.TEST_DB_PROVIDER }}
         run: |
-          bash -x .github/scripts/build.sh
+          tar -xaf cln-${CFG}.tar.bz2
+          poetry run pytest tests/ -vvv -n ${PYTEST_PAR} ${PYTEST_OPTS}
 
-      - name: Upload Unit Test Results
-        if: always()
-        uses: actions/upload-artifact@v2
-        with:
-          name: Junit Report ${{ github.run_number }}.${{ matrix.cfg }}
-          path: report.*
-
-  rust-test:
-    name: Rust Test Config
-    runs-on: ubuntu-20.04
-    needs: [smoke-test]
+  integration-valgrind:
+    name: Valgrind Test CLN ${{ matrix.name }}
+    runs-on: ubuntu-22.04
     env:
+      COMPAT: 1
+      BITCOIN_VERSION: 0.20.1
+      ELEMENTS_VERSION: 0.18.1.8
+      RUST_PROFILE: release  # Has to match the one in the compile step
+      VALGRIND: 1
+      CFG: gcc-dev1-exp1
       DEVELOPER: 1
-      RUST: 1
-      VALGRIND: 0
-      # Run only the rust tests, others are not impacted.
-      TEST_CMD: "make -j 8 && pytest -vvv tests/test_cln_rs.py"
+      EXPERIMENTAL_FEATURES: 1
+      PYTEST_OPTS: --test-group-random-seed=42
```

Comment:
Could you explain the purpose of the random seed? If it's an attempt to achieve better resource balanced between test-groups, I like the concept, but the benefit may be ephemeral as tests are added or as the codebase changes. If this is useful, maybe we could automate this with a cron job to periodically rebalance?  I guess you would quantify time and resource use of each test and then generate random seeds until the balance is relatively equal across test groups.

Reply:
Sure, we can play with the parameters, but it must be the same for all runs. Each group selects it's tests based on this value, and so if we want to run all tests they need to use the same selection criteria.

This can be easily seen when running with `putest-xdist`, i.e., `-n 5` or so. Without specifying the random seed, each process gets to pick it's own, resulting in overlaps in test sets (which xdist catches) but the same issue is true for test groups across multiple ci runners and there we don't have a way to check that tests don't overlap, other than parameterizing them all the same.